### ISSUE TITLE
no-multiple-empty-lines fails on last blank line

### ISSUE THREAD

**Issue by PhilT** [eslint/eslint#2359]
If you set this rule to only allow one blank line, it incorrectly reports the final blank line as failing this rule.

For example:

        1 'use strict';
    >>  2   

My `.eslintrc` file

```
---
  env:
    browser: true
    node: true
    jasmine: true
    phantomjs: true

  globals:
    FB: true
    gon: true
    fbInitialized: true
    config: true

  rules:
    no-cond-assign:
      - 1
      - always
    no-extra-parens: 1
    valid-jsdoc: 1

    block-scoped-var: 1
    complexity:
      - 1
      - 5
    no-param-reassign: 1
    no-self-compare: 1
    no-void: 1
    vars-on-top: 1
    wrap-iife: 1
    strict:
      - 2
      - global

    indent:
      - 2
      - 2
    brace-style:
      - 2
      - 1tbs
      - allowSingleLine: true
    camelcase: 2
    comma-spacing: 2
    comma-style: 2
    consistent-this:
      - 2
      - self
    func-style:
      - 2
      - expression
    max-nested-callbacks:
      - 2
      - 3
    newline-after-var: 2
    no-continue: 2
    no-inline-comments: 2
    no-lonely-if: 2
    no-multiple-empty-lines:
      - 2
      - max: 1
    no-nested-ternary: 2
    no-new-object: 2
    no-trailing-spaces: 2
    operator-assignment:
      - 2
      - always
    operator-linebreak: 2
    padded-blocks:
      - 2
      - never
    quote-props:
      - 2
      - as-needed
    quotes:
      - 2
      - single
      - avoid-escape
    space-after-keywords: 2
    space-before-blocks: 2
    space-before-function-paren:
      - 2
      - anonymous: always
        named: never
    space-in-brackets: 2
    space-in-parens: 2

```

**Comment by nzakas** [comment]
Can you also paste the console output you're seeing?

**Comment by PhilT** [comment]
Sure.

```
test.js
  3:0  warning  Multiple blank lines not allowed  no-multiple-empty-lines

✖ 1 problem (0 errors, 1 warning)

```

**Comment by ilyavolodin** [comment]
Hmm.. I just tried to reproduce this in online demo with the following code:
```js
/* eslint no-multiple-empty-lines: [2, { max: 1}] */
"use strict";
```
And I'm not getting that error. However, I'm also not getting an error if I have two empty trailing linebreaks, which I would expert to get. I'm also getting a wrong behavior from `global-strict`, but that's unrelated to this issue.

**Comment by PhilT** [comment]
Did you add a single blank line at the end?

On my Archlinux commandline I can do:

    echo "/* eslint no-multiple-empty-lines: [2, { max: 1}] */
    > " | eslint --stdin

and get:

    <text>
      3:0  error  Multiple blank lines not allowed  no-multiple-empty-lines

    ✖ 1 problem (1 error, 0 warnings)

And that's with no `.eslintrc` file in the directory.

**Comment by PhilT** [comment]
I also forked the code and wrote a test to check single blank lines but it passes. I wonder if maybe EOF is seen as another blank line.

**Comment by ilyavolodin** [comment]
Could it be CR vs CRLB situation? Maybe you are adding CRLB and it counts as two line breaks somehow?

**Comment by gcochard** [comment]
What version is this happening on? I fixed this in #2458 which was included in `v0.21.0`.

**Comment by PhilT** [comment]
Aha, just upgraded and it's fixed. My apologies.
